ipie/addons/thermal/utils/testing.py

```diff
@@ -0,0 +1,334 @@
+import numpy
+import pytest
+from typing import Union
+
+from ipie.utils.mpi import MPIHandler
+from ipie.utils.testing import generate_hamiltonian
+from ipie.hamiltonians.generic import Generic as HamGeneric
+
+from ipie.addons.thermal.utils.ueg import UEG
+from ipie.addons.thermal.trial.one_body import OneBody
+from ipie.addons.thermal.trial.mean_field import MeanField
+from ipie.addons.thermal.walkers.uhf_walkers import UHFThermalWalkers
+from ipie.addons.thermal.propagation.phaseless_generic import PhaselessGeneric
+from ipie.addons.thermal.qmc.options import ThermalQMCParams
+from ipie.addons.thermal.qmc.thermal_afqmc import ThermalAFQMC
+
+
+def build_generic_test_case_handlers(options: dict,
+                                     seed: Union[int, None],
+                                     debug: bool = False,
+                                     with_eri: bool = False,
+                                     verbose: int = 0):
+    # Unpack options
```

Comment:
I think it might be better to a) explicitly spell out the options in the dict as kwargs, you're not saving anything by unpacking them later, and it serves as form of documentation to have them as args/kwargs OR b) define little dataclasses which group the options so that you don't need to pass in 20 arguments.